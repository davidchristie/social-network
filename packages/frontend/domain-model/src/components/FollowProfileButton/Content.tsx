import { Button } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import {
  FollowProfile,
  FollowProfileVariables,
  UnfollowProfile,
  UnfollowProfileVariables
} from "../../generated/types";

export interface Props {
  account: {
    profile: {
      following: Array<{
        id: string;
      }>;
    };
  };
  followProfile: MutationFn<FollowProfile, FollowProfileVariables>;
  profileId: string;
  unfollowProfile: MutationFn<UnfollowProfile, UnfollowProfileVariables>;
}

const Content: React.StatelessComponent<Props> = ({
  account,
  followProfile,
  profileId,
  unfollowProfile,
}) => {
  const isFollowing = account.profile.following.some(followed => {
    return followed.id === profileId;
  });
  return isFollowing
    ? (
      <Button
        onClick={() => unfollowProfile()}
        variant="contained"
      >
        Unfollow
      </Button>
    )
    : (
      <Button
        onClick={() => followProfile()}
        style="primary"
        variant="contained"
      >
        Follow
      </Button>
    );
};

export default Content;
