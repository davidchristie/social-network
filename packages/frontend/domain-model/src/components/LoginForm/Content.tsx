import { Alert, Button, Input, Section } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import { Link } from "react-router-dom";
import { Login, LoginVariables } from "../../generated/types";

export interface Props {
  login: MutationFn<Login, LoginVariables>;
}

export interface State {
  email: string;
  errorMessage: string | null;
  password: string;
}

export default class Content extends React.Component<Props, State> {
  public state: State = {
    email: "",
    errorMessage: null,
    password: "",
  };

  public render () {
    return (
      <Section className="LoginForm">
        <form onSubmit={this.handleFormSubmit}>
          <h2>Login</h2>
          Don't have an account? <Link to="/signup">Click here to sign up.</Link>
          <hr />
          {this.state.errorMessage && (
            <Alert>{this.state.errorMessage}</Alert>
          )}
          <Input
            autoFocus={true}
            id="login-email"
            label="Email Address"
            name="email"
            onChange={this.handleEmailChange}
            required={true}
            type="email"
            value={this.state.email}
          />
          <Input
            id="login-password"
            label="Password"
            name="password"
            onChange={this.handlePasswordChange}
            required={true}
            type="password"
            value={this.state.password}
          />
          <Button type="submit">Submit</Button>
        </form>
      </Section>
    );
  }

  private handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      email: event.target.value,
    });
  }

  private handleFormSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    const { login } = this.props;
    const { email, password } = this.state;
    try {
      await login({
        variables: {
          email,
          password,
        },
      });
    } catch (error) {
      this.setState({
        email: "",
        errorMessage: "Incorrect email or password.",
        password: "",
      });
    }
  }

  private handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      password: event.target.value,
    });
  }
}
