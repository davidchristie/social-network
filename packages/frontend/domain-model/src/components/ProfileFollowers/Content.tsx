import { Avatar, Grid, Section } from "design-system";
import React from "react";
import { Link } from "react-router-dom";

export interface Props {
  profile: {
    followers: Array<{
      avatar: {
        url: string;
      } | null;
      id: string;
      name: string;
    }>;
  };
}

export interface State {
  anchorElement: HTMLElement | null;
}

export default class Content extends React.Component<Props, State> {
  public render () {
    const { profile } = this.props;
    return (
      <Section className="ProfileFollowers">
        <h2>Followers</h2>
        <Grid container={true} >
          {profile.followers.map(({ avatar, id, name }) => {
            return (
              <Grid item={true} key={id} sm={4} xs={12}>
                <div className="ProfileFollowers__profile">
                  <Avatar
                    image={avatar ? avatar.url : undefined}
                    size="small"
                  />
                  <Link to={`/profile/${id}`}>
                    {name}
                  </Link>
                </div>
              </Grid>
            );
          })}
        </Grid>
      </Section>
    );
  }
}
