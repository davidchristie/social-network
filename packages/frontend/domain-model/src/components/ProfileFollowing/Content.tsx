import { Avatar, Section } from "design-system";
import React from "react";
import { Link } from "react-router-dom";

export interface Props {
  profile: {
    following: Array<{
      avatar: {
        url: string;
      } | null;
      id: string;
      name: string;
    }>;
  };
}

export interface State {
  anchorElement: HTMLElement | null;
}

export default class Content extends React.Component<Props, State> {
  public render () {
    const { profile } = this.props;
    return (
      <Section className="ProfileFollowing">
        <h2>Following</h2>
        {profile.following.map(({ avatar, id, name }) => {
          return (
            <div className="ProfileFollowing__profile" key={id}>
              <Avatar
                image={avatar ? avatar.url : undefined}
                size="small"
              />
              <Link to={`/profile/${id}`}>
                {name}
              </Link>
            </div>
          );
        })}
      </Section>
    );
  }
}
