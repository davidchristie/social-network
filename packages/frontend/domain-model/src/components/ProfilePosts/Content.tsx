import React from "react";
import { PostSummary as PostSummaryData } from "../../generated/types";
import PostSummary from "../PostSummary";

export interface Props {
  profile: {
    posts: PostSummaryData[]
  };
}

const Content: React.StatelessComponent<Props> = ({ profile }) => {
  return (
    <div className="ProfilePosts">
      {
        profile.posts
          .map(post => <PostSummary key={post.id} post={post} />)
          .reverse()
      }
    </div>
  );
};

export default Content;
