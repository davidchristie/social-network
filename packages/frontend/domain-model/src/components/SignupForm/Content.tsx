import { Alert, Button, Input, Section } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import { Link } from "react-router-dom";
import { Signup, SignupVariables } from "../../generated/types";

export interface Props {
  signup: MutationFn<Signup, SignupVariables>;
}

export interface State {
  email: string;
  errorMessage: string | null;
  name: string;
  password: string;
}

export default class Content extends React.Component<Props, State> {
  public state: State = {
    email: "",
    errorMessage: null,
    name: "",
    password: "",
  };

  public render () {
    return (
      <Section className="SignupForm">
        <form onSubmit={this.handleFormSubmit}>
          <h2>Signup</h2>
          Already have an account? <Link to="/login">Click here to log in.</Link>
          <hr />
          {this.state.errorMessage && <Alert>{this.state.errorMessage}</Alert>}
          {this.renderNameInput()}
          {this.renderEmailInput()}
          {this.renderPasswordInput()}
          <Button type="submit">Submit</Button>
        </form>
      </Section>
    );
  }

  private handleFormSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    try {
      await this.props.signup({
        variables: {
          email: this.state.email,
          name: this.state.name,
          password: this.state.password,
        },
      });
    } catch (error) {
      this.setState(previousState => ({
        ...previousState,
        email: "",
        errorMessage: error.message,
        password: "",
      }));
    }
  }

  private handleInputChange = (name: "email" | "name" | "password") => {
    return (event: React.ChangeEvent<HTMLInputElement>) => {
      event.persist();
      this.setState(previousState => ({
        ...previousState,
        [name]: event.target.value,
      }));
    };
  }

  private renderEmailInput = () => {
    return (
      <Input
        id="signup-email"
        label="Email Address"
        name="email"
        onChange={this.handleInputChange("email")}
        required={true}
        type="email"
        value={this.state.email}
      />
    );
  }

  private renderNameInput = () => {
    return (
      <Input
        autoFocus={true}
        id="signup-name"
        label="Name"
        name="name"
        onChange={this.handleInputChange("name")}
        required={true}
        value={this.state.name}
      />
    );
  }

  private renderPasswordInput = () => {
    return (
      <Input
        id="signup-password"
        label="Password"
        name="password"
        onChange={this.handleInputChange("password")}
        required={true}
        type="password"
        value={this.state.password}
      />
    );
  }
}
