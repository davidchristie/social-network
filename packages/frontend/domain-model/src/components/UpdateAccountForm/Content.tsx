import { Alert, Button, Input, Section } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import {
  UpdateAccountData,
  UpdateAccountVariables,
} from "../../mutations/UpdateAccount";

export interface Props {
  account: {
    email: string;
  };
  updateAccount: MutationFn<UpdateAccountData, UpdateAccountVariables>;
}

export interface State {
  email: string;
  errorMessages: string[];
}

export default class UpdateAccountForm extends React.Component<Props, State> {
  public state: State = {
    email: this.props.account.email,
    errorMessages: [],
  };

  public render () {
    const { updateAccount } = this.props;
    const { email, errorMessages } = this.state;
    return (
      <Section className="UpdateAccountForm">
        <h2>Account</h2>
        <form
          onSubmit={async (event: React.FormEvent) => {
            event.preventDefault();
            const result = await updateAccount({
              variables: {
                data: {
                  email,
                },
              },
            });
            if (result && result.errors) {
              this.setState({
                errorMessages: result.errors.map(error => error.message),
              });
            }
          }}
        >
          {errorMessages.map((errorMessage, index) => (
            <Alert key={index}>{errorMessage}</Alert>
          ))}
          <Input
            id="update-account-email"
            label="Email Address"
            name="email"
            onChange={this.emailChanged}
            required={true}
            type="email"
            value={this.state.email}
          />
          <Button type="submit">Save</Button>
        </form>
      </Section>
    );
  }

  private emailChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      email: event.target.value,
    });
  }
}
