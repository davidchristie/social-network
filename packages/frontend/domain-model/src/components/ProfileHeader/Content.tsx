import { Avatar, Section } from "design-system";
import React from "react";
import FollowProfileButton from "../FollowProfileButton";

export interface Props {
  account: {
    profile: {
      id: string;
    };
  } | null;
  profile: {
    avatar: {
      url: string;
    } | null;
    id: string;
    name: string;
  };
}

const Content: React.StatelessComponent<Props> = ({ account, profile }) => {
  const isAuthenticated = Boolean(account);
  const isOwnProfile = account && account.profile.id === profile.id;
  return (
    <Section>
      <Avatar
        image={profile.avatar ? profile.avatar.url : undefined}
        size="large"
      />
      <h1>{profile.name}</h1>
      {isAuthenticated && !isOwnProfile && (
        <FollowProfileButton profileId={profile.id} />
      )}
    </Section>
  );
};

export default Content;
