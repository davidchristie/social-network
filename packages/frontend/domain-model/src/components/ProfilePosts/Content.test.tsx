import { describeWithProps } from "test-utilities/react";
import mockProfileResponse from "../../queries/mockProfileResponse";
import Content from "./Content";

const profile = mockProfileResponse({
  id: "profile_id",
}).result!.data!.profile;

describe("Content component", () => {
  describeWithProps("with posts", Content, {
    profile,
  });
});
