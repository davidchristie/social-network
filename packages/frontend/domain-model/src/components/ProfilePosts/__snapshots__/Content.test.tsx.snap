// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component with posts matches snapshot 1`] = `
<div
  className="ProfilePosts"
>
  <PostSummary
    key="post_id_2"
    post={
      Object {
        "__typename": "Post",
        "createdAt": "2018-10-22T01:15:47.347Z",
        "createdBy": Object {
          "__typename": "Profile",
          "avatar": Object {
            "__typename": "Image",
            "id": "avatar_id",
            "url": "https://avatar_url.com",
          },
          "id": "profile_id",
          "name": "Profile",
        },
        "id": "post_id_2",
        "text": "Post 2",
      }
    }
  />
  <PostSummary
    key="post_id_1"
    post={
      Object {
        "__typename": "Post",
        "createdAt": "2018-10-22T01:15:47.347Z",
        "createdBy": Object {
          "__typename": "Profile",
          "avatar": Object {
            "__typename": "Image",
            "id": "avatar_id",
            "url": "https://avatar_url.com",
          },
          "id": "profile_id",
          "name": "Profile",
        },
        "id": "post_id_1",
        "text": "Post 1",
      }
    }
  />
  <PostSummary
    key="post_id_0"
    post={
      Object {
        "__typename": "Post",
        "createdAt": "2018-10-22T01:15:47.347Z",
        "createdBy": Object {
          "__typename": "Profile",
          "avatar": Object {
            "__typename": "Image",
            "id": "avatar_id",
            "url": "https://avatar_url.com",
          },
          "id": "profile_id",
          "name": "Profile",
        },
        "id": "post_id_0",
        "text": "Post 0",
      }
    }
  />
</div>
`;
